src/findCollisions.js

```diff
@@ -0,0 +1,27 @@
+
+module.exports = function findCollisions (timetable, constraint) {
```

Comment:
This seems overly complicated.
What this PR is providing is essentially the ability for a user to provide a function which is being run instead of `findCollisions` for certain constraint.

We can just keep the old code and add at the beginning:
```
if(typeof constraint === "function") return constraint(timetable)
```